### ISSUE TITLE
In orthographic projection, plots on other side of globe plotted at pole of projection (when plotted one at a time)

### ISSUE THREAD

**Issue by ramzim** [SciTools/cartopy#1928]
### Description

In version 0.17.0 with matplotlib version 3.3.2...

When plotting points one at a time, those that should not be plotted are returned at the pole of the projection. This is not the case for a set of points all plotted together. See the example on SO: https://stackoverflow.com/questions/66216852/cartopy-plotting-extra-points-at-poles-when-passed-one-at-a-time

#### Code to reproduce

```
import numpy as np
import matplotlib.pyplot as plt
import cartopy.crs as ccrs

projection = ccrs.Orthographic(0, 90)
transform = ccrs.Geodetic()

fig = plt.figure()
ax = fig.add_subplot(1, 1, 1, projection = projection)

ax.coastlines()
ax.set_global()
ax.gridlines()

npoints = 100
np.random.seed(71077345)
lon = np.random.sample(npoints) * 360
lat = np.random.sample(npoints) * 180 - 90
plt.plot(lon,
         lat,
         'ro',
         alpha = 0.3,
         transform = transform)

for i in range(npoints):
    plt.plot(lon[i],
             lat[i],
             'b.',
             alpha = 0.3,
             transform = transform)
plt.show()
```

![points_at_pole](https://user-images.githubusercontent.com/11670408/141368638-a302681c-42a3-4dcc-beda-cd3ff66df73e.png)

#### Traceback 

```

```

<details>
 <summary>Full environment definition</summary>

### Operating system
Windows

### Cartopy version
0.17.0

### conda list
matplotlib 3.3.2
```
```

### pip list

```
```

</details>

**Comment by dopplershift** [comment]
I am not able to reproduce this with Matplotlib 3.4.3 and CartoPy 0.20.1 (Conda-forge Python 3.9):

<img width="231" alt="image" src="https://user-images.githubusercontent.com/221526/141369119-13424676-5f4b-4104-a603-77825dc897fa.png">

**Comment by greglucas** [comment]
Should be fixed in v0.19 and beyond by https://github.com/SciTools/cartopy/pull/1710
Feel free to reopen if this didn't solve the issue.

**Comment by ramzim** [comment]
Thanks for the quick reply. I will update my modules.
